### ISSUE TITLE
list archive stylesheets need modernization

### ISSUE THREAD

**Issue by gosko** [w3c/mailing-list-archives#1]
The style of [W3C's list archives](https://lists.w3.org/Archives/Public/) was last revised 12+ years ago and lacks readability.

Among other things, it would be nice if the text were displayed in a reflowable proportional font by default.

I put a copy of our current stylesheets in the [style directory](../tree/master/style/) in this repo; improvements welcome. (though we will likely want to overhaul the HTML as well)

**Comment by r12a** [comment]
for the body text of emails, i looked at various existing approaches and experimented with a number of things while developing the thread flattener at http://www.w3.org/Mail/flatten/

some sites tried to do clever things, like back running together lines that had been chopped to a certain length, but none really seemed to pull it off completely.

in the end, i settled for a div with CSS set to 
white-space: pre-wrap;
overflow: auto;
...

that seems to work pretty well.

**Comment by gosko** [comment]
I made a [mockup of a conversion to HTML5+bootstrap](https://w3c.github.io/mailing-list-archives/samples/message-proposal-1.html) as one possible version of an updated message page; feedback welcome.

Based on this experience I don't think we should put any effort into updating the old stylesheets as is, but simply scrap them and start from scratch.

I wonder what people think about the choice of bootstrap for use with our archives? I know a few of us on W3C's systems team have favorable impressions of bootstrap, though we don't have a lot of experience with it yet. It seems like it can provide a lot of what we need in terms of responsive and accessible markup, and has the advantage of being widely reviewed and tested.

**Comment by gosko** [comment]
@r12a wrote:

> some sites tried to do clever things, like back running together lines that had been chopped to a certain length, but none really seemed to pull it off completely.

I think any attempt at clever formatting of email will have mixed results due to the huge variety of email formats in use, so it should be easy for users to disable any smart formatting to view the message in its original format.

Maybe it would be best to continue to generate messages in a boring but predictable format as currently done by hypermail, and use a javascript library to apply intelligent formatting?

A separate library would make it easier to evolve the formatting smarts independently from our archives, which are a [nuisance to regenerate](https://github.com/w3c/mailing-list-archives/issues/3).  It could even allow users to set their own preferences for how messages are displayed.

**Comment by marcoscaceres** [comment]
Have we done any experimentation with just setting `<meta name="viewport" content="width=device-width, initial-scale=1.0">`? I mean, that would already be a massive potential improvement over the current situation.

**Comment by tripu** [comment]
+1 to @marcoscaceres idea; I'll try to submit a PR with that, and a few other easy, generic enhancements.

> A separate library would make it easier to evolve the formatting smarts independently from our archives, which are a [nuisance to regenerate](https://github.com/w3c/mailing-list-archives/issues/3).  It could even allow users to set their own preferences for how messages are displayed.

Yes, I think our attempts to be clever and reformat messages should be ideally encapsulated in a separate library. Among other things, to allow us (and users) to switch those features on and off easily (that's why I'd prefer to do that with JS). I didn't do it that way when I submitted #5, but I'll rework that&hellip;

**Comment by gosko** [comment]
> Have we done any experimentation with just setting <meta name="viewport" content="width=device-width, initial-scale=1.0">? I mean, that would already be a massive potential improvement over the current situation.

@marcoscaceres yes, that's included in [my proposal](https://w3c.github.io/mailing-list-archives/samples/message-proposal-1.html), feedback welcome!
